pynamodb/models.py

```diff
@@ -169,40 +171,104 @@ class MetaModel(AttributeContainerMeta):
+        parent_meta_class = None
+        is_parent_abstract = None
+        if parent  and not parent_is_attribute_container:
+            parent_meta_class = getattr(parent, META_CLASS_NAME, None)
+        if parent_meta_class and hasattr(parent_meta_class, ABSTRACT):
+            is_parent_abstract = getattr(parent_meta_class, ABSTRACT)
+
+        # Temporary check until implementing other inheritance types
+        if parent and not parent_is_attribute_container and not parent_is_model and not is_parent_abstract:
+            raise NotImplementedError("Inheritance from non abstract models is not supported now")
+
         if isinstance(attrs, dict):
+            meta_class = attrs.get(META_CLASS_NAME, None)
+
+            # Check special parents to avoid copying DefaultMeta
+            if parent and not parent_is_attribute_container and not parent_is_model:
+                if not meta_class:
+                    # Prepare an empty meta class
+                    meta_class = type(META_CLASS_NAME, (), {})
+                    setattr(cls, META_CLASS_NAME, meta_class)
+
+                if parent_meta_class:
+                        # Copying missing meta attributes from parent(s)
+                        for item in dir(parent_meta_class):
+                            # Avoid both private and existant attributes
+                            if not item.startswith("__") and not hasattr(meta_class, item):
+                                setattr(meta_class, item, copy.deepcopy(getattr(parent_meta_class, item)))
+
+            is_abstract = None
+            has_table_name = False
+            has_indexes = False
+            if meta_class:
+                if not hasattr(meta_class, REGION):
+                    setattr(meta_class, REGION, get_settings_value('region'))
+                if not hasattr(meta_class, HOST):
+                    setattr(meta_class, HOST, get_settings_value('host'))
+                if not hasattr(meta_class, 'session_cls'):
+                    setattr(meta_class, 'session_cls', get_settings_value('session_cls'))
+                if not hasattr(meta_class, 'request_timeout_seconds'):
+                    setattr(meta_class, 'request_timeout_seconds', get_settings_value('request_timeout_seconds'))
+                if not hasattr(meta_class, 'base_backoff_ms'):
+                    setattr(meta_class, 'base_backoff_ms', get_settings_value('base_backoff_ms'))
+                if not hasattr(meta_class, 'max_retry_attempts'):
+                    setattr(meta_class, 'max_retry_attempts', get_settings_value('max_retry_attempts'))
+                if not hasattr(meta_class, ABSTRACT):
+                    setattr(meta_class, ABSTRACT, False)
+                is_abstract = getattr(meta_class, ABSTRACT, None)
+                if hasattr(meta_class, 'table_name'):
+                    has_table_name = True
+            else:
+                # TODO: decide what to do with DefaultMeta
```

Comment:
should we ban this behaviour? Currently I left it allowed.

Reply:
Note: if we ban this behavior, this would be a breaking change and we'd have to release a new major version of Pynamo.